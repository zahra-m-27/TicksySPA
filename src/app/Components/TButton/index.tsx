import React from 'react';
import styles from './styles.module.scss';
import ClassNames from '../../Utilities/ClassNames';

interface Props {
  label?: string;
  className?: string;
  onClick?: () => void;
  labelClassName?: string;
  backgroundColor?: string;
}

export default function TButton({
  label,
  onClick,
  className,
  labelClassName,
  backgroundColor,
}: Props) {
  return (
    <div
      className={ClassNames(styles.button, className)}
      onClick={onClick}
      style={{backgroundColor}}>
      <span className={ClassNames(styles.label, labelClassName)}>{label}</span>
    </div>
  );
}
