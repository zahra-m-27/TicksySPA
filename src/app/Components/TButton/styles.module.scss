.button {
  display: flex;
  color: #ffffff;
  cursor: pointer;
  padding: 10px 20px;
  border-radius: 6px;
  align-items: center;
  transition: all 0.5s;
  justify-content: center;
  background-color: #0088e3;

  &:hover {
    opacity: 0.8;
  }

  .label {
    user-select: none;
  }
}
