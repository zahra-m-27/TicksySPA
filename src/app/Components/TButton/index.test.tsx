import ReactDOM from 'react-dom';
import React from 'react';
import TButton from '.';
import {render} from '@testing-library/react';
import user from '@testing-library/user-event';
import renderer from 'react-test-renderer';

const onClick = jest.fn();

const sample = (
  <TButton label="Test Label" onClick={onClick} backgroundColor="#232323" />
);

test('render in ReactDOM', () => {
  const div = document.createElement('div');
  ReactDOM.render(sample, div);
});

it('render label', () => {
  const {getByText} = render(sample);
  getByText(/Test Label/i);
});

it('render background', () => {
  const {getByText} = render(sample);
  const labelElement = getByText(/Test Label/i);
  expect(
    labelElement.parentElement?.style.backgroundColor === 'rgb(35, 35, 35)'
  ).toBeTruthy();
});

it('button click', () => {
  const {getByText} = render(sample);
  const buttonElement = getByText(/Test Label/i);
  expect(onClick).toBeCalledTimes(0);
  user.click(buttonElement);
  expect(onClick).toBeCalledWith(expect.anything());
  expect(onClick).toBeCalledTimes(1);
});

it('snapshot', () => {
  const snapshot = renderer.create(sample).toJSON();
  expect(snapshot).toMatchSnapshot();
});
