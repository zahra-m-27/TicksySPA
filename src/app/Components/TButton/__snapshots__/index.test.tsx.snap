// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="button"
  onClick={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#232323",
    }
  }
>
  <span>
    Test Label
  </span>
</div>
`;
